homeassistant/components/reolink/light.py

```diff
@@ -0,0 +1,157 @@
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from . import ReolinkData
+from .const import DOMAIN
+from .entity import ReolinkCoordinatorEntity
+
+
+@dataclass
+class ReolinkLightEntityDescriptionMixin:
+    """Mixin values for Reolink light entities."""
+
+    is_on_fn: Callable[[Host, int | None], bool]
+    turn_on_off_fn: Callable[[Host, int | None, bool], Any]
+
+
+@dataclass
+class ReolinkLightEntityDescription(
+    LightEntityDescription, ReolinkLightEntityDescriptionMixin
+):
+    """A class that describes light entities."""
+
+    supported_fn: Callable[[Host, int | None], bool] = lambda api, ch: True
+    get_brightness_fn: Callable[[Host, int | None], int] | None = None
+    set_brightness_fn: Callable[[Host, int | None, float], Any] | None = None
+
+
+LIGHT_ENTITIES = (
+    ReolinkLightEntityDescription(
+        key="floodlight",
+        name="Floodlight",
+        icon="mdi:spotlight-beam",
+        supported_fn=lambda api, ch: api.supported(ch, "floodLight"),
+        is_on_fn=lambda api, ch: api.whiteled_state(ch),
+        turn_on_off_fn=lambda api, ch, value: api.set_whiteled(ch, state=value),
+        get_brightness_fn=lambda api, ch: api.whiteled_brightness(ch),
+        set_brightness_fn=lambda api, ch, value: api.set_whiteled(
+            ch, brightness=int(value)
+        ),
+    ),
+    ReolinkLightEntityDescription(
+        key="ir_lights",
+        name="Infra red lights in night mode",
+        icon="mdi:led-off",
+        supported_fn=lambda api, ch: api.supported(ch, "ir_lights"),
+        is_on_fn=lambda api, ch: api.ir_enabled(ch),
+        turn_on_off_fn=lambda api, ch, value: api.set_ir_lights(ch, value),
+    ),
+    ReolinkLightEntityDescription(
+        key="status_led",
+        name="Status LED",
+        icon="mdi:lightning-bolt-circle",
+        supported_fn=lambda api, ch: api.supported(ch, "status_led"),
+        is_on_fn=lambda api, ch: api.status_led_enabled(ch),
+        turn_on_off_fn=lambda api, ch, value: api.set_status_led(ch, value),
+    ),
+)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up a Reolink light entities."""
+    reolink_data: ReolinkData = hass.data[DOMAIN][config_entry.entry_id]
+
+    async_add_entities(
+        ReolinkLightEntity(reolink_data, channel, entity_description)
+        for entity_description in LIGHT_ENTITIES
+        for channel in reolink_data.host.api.channels
+        if entity_description.supported_fn(reolink_data.host.api, channel)
+    )
+
+
+class ReolinkLightEntity(ReolinkCoordinatorEntity, LightEntity):
+    """Base light entity class for Reolink IP cameras."""
+
+    entity_description: ReolinkLightEntityDescription
+
+    def __init__(
+        self,
+        reolink_data: ReolinkData,
+        channel: int,
```

Comment:
True, this was because I was thinking about NVR entities that would not have a channel, but I ended up using seperate EntityDescriptions for that like in PR https://github.com/home-assistant/core/pull/87943/files

I have changed the typing to only `int` across all 6 PRs.